src/source.rs

```diff
@@ -286,6 +303,72 @@ impl ImageLocation for StreamLocation {
     }
 }
 
+impl OsmetLocation {
+    pub fn new(architecture: &str, sector_size: u32) -> Result<Option<Self>> {
+        let osmet_dir = Path::new(OSMET_FILES_DIR);
+        if !osmet_dir.exists() {
+            return Ok(None);
+        }
+
+        if let Some((osmet_path, description)) =
+            find_matching_osmet_in_dir(osmet_dir, architecture, sector_size)?
+        {
+            Ok(Some(Self {
+                osmet_path,
+                architecture: architecture.into(),
+                sector_size,
+                description,
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+impl Display for OsmetLocation {
+    fn fmt(&self, f: &mut Formatter<'_>) -> ::std::fmt::Result {
+        write!(
+            f,
+            "Installing {} {} ({}-byte sectors)",
+            self.description, self.architecture, self.sector_size
+        )
+    }
+}
+
+impl ImageLocation for OsmetLocation {
+    fn sources(&self) -> Result<Vec<ImageSource>> {
+        let unpacker = OsmetUnpacker::new_from_sysroot(Path::new(&self.osmet_path))?;
+
+        let filename = {
+            let mut filename = self.osmet_path.clone();
```

Comment:
This is the entire path, right?  The field should have just the filename.

Reply:
Man, I should've just stuck with `image.raw` for this... ;)  Anyway, I went the whole way with this now and it somehow spans 18 lines post-`cargo fmt`. :tada: 

(Though I do think in the future it'd be nice to have `download` be able to get back the raw image without actually writing it to a device, so this would be useful then.)

Reply:
Yeah... :disappointed:  Thanks for doing this.